static/app/components/clipboard.tsx

```diff
@@ -58,34 +71,35 @@ class Clipboard extends Component<Props> {
       addSuccessMessage(successMessage);
     }
     onSuccess?.();
-  };
-
-  handleMount = (ref: HTMLElement) => {
-    if (!ref) {
-      return;
-    }
-
-    // eslint-disable-next-line react/no-find-dom-node
-    this.element = findDOMNode(ref);
-    this.element?.addEventListener('click', this.handleClick);
-  };
-
-  render() {
-    const {children, hideUnsupported} = this.props;
-
-    // Browser doesn't support `execCommand`
-    if (hideUnsupported && !isSupported()) {
-      return null;
-    }
-
-    if (!isValidElement(children)) {
-      return null;
-    }
+  }, [value, onError, onSuccess, errorMessage, successMessage, hideMessages]);
+
+  const attachClickHandler = useCallback(() => {
+    elementRef.current?.addEventListener('click', handleClick);
+    return () => elementRef.current?.removeEventListener('click', handleClick);
+  }, [handleClick]);
+
+  // Handle click handler clearnup / reattachment
+  useEffect(() => attachClickHandler(), [attachClickHandler]);
```

Comment:
Not 100% positive, but I changed it so that the addEventListener and removeEventListener were coupled in this use effect, since if any of the dependencies change on the handleClick it would remove the event listener, and in the old world would only re-attach when the child element ref is re-assigned (which may never happen).

So now the addEventListener will be called if the `handleClick` updates when it's dependencies change.